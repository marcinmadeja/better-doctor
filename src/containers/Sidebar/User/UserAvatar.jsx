// @flow strict
import * as React from 'react';
import styled, { css } from 'styled-components';
import { withStyles } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import { COLORS } from 'config/theme';

const StyledUserAvatar = styled.div`
  position: relative;
  display: inline-block;
`;

type StatusProps = {
  status?: string,
};

const Status = (styled.div`
  width: 11px;
  height: 11px;
  border-radius: 100%;
  position: absolute;
  bottom: 3px;
  right: 3px;

  ${props => props.status === 'active' && css`
    background-color: ${COLORS.success};
  `}

  ${props => props.status === 'inactive' && css`
    background-color: ${COLORS.failure};
  `}
`: React.ComponentType<StatusProps>);

const styles = {
  bigAvatar: {
    width: 50,
    height: 50,
  },
};

type Props = {
  src: string,
  alt?: string,
  status: string,
  classes: { [string]: string },
};

const UserAvatar = ({ src, alt, status, classes }: Props) => {
  return (
    <StyledUserAvatar>
      <Avatar src={src} alt={alt} className={classes.bigAvatar} />

      <Status status={status} />
    </StyledUserAvatar>
  );
};

UserAvatar.defaultProps = {
  alt: '',
};

export default withStyles(styles)(UserAvatar);
