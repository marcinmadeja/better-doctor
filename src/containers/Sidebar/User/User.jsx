// @flow strict
import * as React from 'react';
import styled from 'styled-components';
import { USER, GUTTER, THEME_COLORS } from 'constants/index';
import UserAvatar from './UserAvatar';

const UserWrap = styled.div`
  display: flex;
  align-items: center;
  padding: ${GUTTER / 2}px ${GUTTER / 2}px;
  background-color: ${THEME_COLORS.darkLighter};
  font-size: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const Name = styled.div`
  margin-left: 15px;
  color: ${THEME_COLORS.darkContrast};
  letter-spacing: 0.7px;
`;

class User extends React.Component<{}> {
  render() {
    const userName = `${USER.name.first} ${USER.name.last}`;

    return (
      <UserWrap>
        <UserAvatar
          src={USER.avatar}
          alt={userName}
          status={USER.status}
        />

        <Name>
          {userName}
        </Name>
      </UserWrap>
    );
  }
}

export default User;
