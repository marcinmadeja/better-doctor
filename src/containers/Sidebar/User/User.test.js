import React from 'react';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router';
import User from './User';

const defaultProps = {};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const actions = {};
  const component = <User {...actions} {...props} />;
  const memoryComponent = (
    <MemoryRouter initialEntries={['/']}>
      {component}
    </MemoryRouter>
  );
  const shallowComponent = shallow(component);

  return {
    actions,
    component,
    shallowComponent,
    memoryComponent,
  };
};

describe('User', () => {
  it('should render', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });

  it('should match basic snapshot', () => {
    const { memoryComponent } = setup();

    const tree = renderer
      .create(memoryComponent)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
