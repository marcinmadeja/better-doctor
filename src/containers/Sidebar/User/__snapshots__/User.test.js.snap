// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should match basic snapshot 1`] = `
.c1 {
  position: relative;
  display: inline-block;
}

.c2 {
  width: 11px;
  height: 11px;
  border-radius: 100%;
  position: absolute;
  bottom: 3px;
  right: 3px;
  background-color: #0aa699;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12.5px 12.5px;
  background-color: #22262e;
  font-size: 18px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3 {
  margin-left: 15px;
  color: #ffffff;
  -webkit-letter-spacing: 0.7px;
  -moz-letter-spacing: 0.7px;
  -ms-letter-spacing: 0.7px;
  letter-spacing: 0.7px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="MuiAvatar-root-2 UserAvatar-bigAvatar-1"
    >
      <img
        alt="John Smith"
        className="MuiAvatar-img-4"
        sizes={undefined}
        src="avatar-photo.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="c2"
    />
  </div>
  <a
    href="/account"
    onClick={[Function]}
  >
    <div
      className="c3"
    >
      John Smith
    </div>
  </a>
</div>
`;
